### PR TITLE
Feat/testnet

### PR DESCRIPTION
**Pull request by raop155**
# Description

Fixes 115

- Add `network dropdown` to change between `mainnet` and `testnet` networks.

image

### CODE REVIEW

File: src/components/molecules/Header/index.tsx [hunk 1]
```diff
@@ -32,8 +35,27 @@ const HeaderLinks = () => (
...
+  { label: "Testnet", value: "testnet" },
+];
+
+const getCurrentNetworkItem = (network: NETWORK): NetworkSelectProps => {
+  return NETWORK_LIST.find(item => item.value === network) || NETWORK_LIST[0];
+};
+
+const Header = ({ network }: Props) => {
   const { t } = useTranslation();
+  const [, setSearchParams] = useSearchParams();
```
Comment (yuli-ferna) [inline src/components/molecules/Header/index.tsx:55]: Remove the comma

**Review by yuli-ferna — APPROVED** [review]

File: src/components/molecules/TransactionHistoryChart/index.tsx [hunk 2]
```diff
@@ -17,49 +17,48 @@ const TransactionHistoryChart = ({ range }: Props) => {
   const [totalTxs, setTotalTxs] = useState("");
   const tickAmount = range === "month" ? 4 : 5;
 
-  const { isLoading, isError, mutate } = useMutation(
-    () => client.guardianNetwork.getLastTxs(range),
-    {
-      onSuccess: response => {
```
Comment (valentinoConti) [inline src/components/molecules/TransactionHistoryChart/index.tsx:23]: it looks like this is doing the same than before but using `useQuery` instead of `useMutation` and processing the data on a useEffect instead of doing it on `onSuccess`

why was this changed? something was working badly? 

also, I think the `useMutation` its being imported but not used now 🙏🏼

**Review by valentinoConti — APPROVED** [review]

File: src/components/molecules/Header/index.tsx (hunk 1 above)
Comment (raop155) [reply src/components/molecules/Header/index.tsx:55]: Yeah I need to use the `comma(,)` here because `useSearchParams()` hook returns me an array and I am only using the `set` method but not the `value` (like `useState` hook) so in order to avoid getting the `value` I just ignore with a `comma(,)`

File: src/components/molecules/TransactionHistoryChart/index.tsx (hunk 2 above)
Comment (raop155) [reply src/components/molecules/TransactionHistoryChart/index.tsx:23]: - When change from `mainnet` to `testnet` network or vice-versa I run the `queryClient.resetQueries() [built-in]` method, this method re-fetch all the active queries, but only the queries that use `useQuery` that's why I changed it in the `TxHistory` and also in the `XChain Activity`.
** keeping `useMutation`: This could also be handled by implementing an `useEffect` and checking if the network change and call the mutation again, etc, but preferred to take with built-in features on the module react-query**


- `onSuccess, onError, etc` on `useQuery` will be deprecated and "considered" a bad practice even tho in the current version that we use is "ok" to use and works. I leave you some refs that I found when researching for this :)

refs: 
https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose
https://twitter.com/t3dotgg/status/1647453694286508033#fromHistory

Comment (valentinoConti) [reply src/components/molecules/TransactionHistoryChart/index.tsx:23]: Excellent! Thank you for the clarifications 💪🏼👌🏼
